### PR TITLE
en2endtest: extend annonelection

### PR DESCRIPTION
**Pull request by mariajdab**
Adding a test with `TempSIKs` flag set to true to vote with half of the accounts that are not registered using `RegisterSIKForVote`, and the remaining half voting with registered annon accounts.
#1083

### CODE REVIEW

File: cmd/end2endtest/helpers.go [hunk 1]
```diff
@@ -655,3 +667,38 @@ func (t *e2eElection) endElectionAndFetchResults() (*vapi.ElectionResults, error
...
+			if _, _, err := t.createAccount(pKey.String()); err != nil &&
+				!strings.Contains(err.Error(), "createAccountTx: account already exists") {
+				errorChan <- err
+			}
+			log.Infow("anonymous census account created", "index", i, "address", acc.AddressString())
+		}(i, acc) // Pass the acc variable as a parameter to avoid data race
+	}
+
+	wg.Wait()
+	close(errorChan) // Close the error channel after all goroutines have finished
```
Comment (altergui) [inline cmd/end2endtest/helpers.go]: this needs to be in a goroutine

File: cmd/end2endtest/helpers.go [hunk 2]
```diff
@@ -655,3 +667,38 @@ func (t *e2eElection) endElectionAndFetchResults() (*vapi.ElectionResults, error
 	}
 	return results, nil
 }
+
+func (t *e2eElection) registerAnnonAccts(voterAccounts []*ethereum.SignKeys) error {
+	errorChan := make(chan error) // Create a buffered channel to prevent deadlock
```
Comment (altergui) [inline cmd/end2endtest/helpers.go]: this is unbuffered, delete comment

**Review by altergui — COMMENTED** [review]
please correct Annon -> Anon

File: cmd/end2endtest/helpers.go (hunk 1 above)
Comment (mariajdab) [reply cmd/end2endtest/helpers.go]: Yeah, but the PR #1103 resolved that

**Comment by mariajdab** [comment]
ready to review @altergui  :raised_hands:

**Review by lucasmenendez — APPROVED** [review]

File: cmd/end2endtest/zkweighted.go [hunk 3]
```diff
@@ -17,13 +17,18 @@ func init() {
 		description: "Performs a complete test of anonymous election, from creating a census to voting and validating votes",
 		example:     os.Args[0] + " --operation=anonelection --votes=1000",
 	}
+	ops["anonelectionTempSIKS"] = operation{
```
Comment (altergui) [inline cmd/end2endtest/zkweighted.go]: i'd rather keep the case consistent, TempSIKs instead of TempSIKS
(there are more ocurrences of this)

**Review by altergui — APPROVED** [review]
LGTM :+1: merge as-is (or fix the case and merge, no need to re-review)

File: cmd/end2endtest/zkweighted.go (hunk 3 above)
Comment (mariajdab) [reply cmd/end2endtest/zkweighted.go]: added the suggestion :raised_hands:

**Pull request merged**